keycloak/openid_client_scope.go

```diff
@@ -51,3 +51,35 @@ func (keycloakClient *KeycloakClient) UpdateOpenidClientScope(clientScope *Openi
 func (keycloakClient *KeycloakClient) DeleteOpenidClientScope(realmId, id string) error {
 	return keycloakClient.delete(fmt.Sprintf("/realms/%s/client-scopes/%s", realmId, id))
 }
+
+func (keycloakClient *KeycloakClient) listOpenidClientScopesWithFilter(realmId string, filter func(*OpenidClientScope) bool) ([]*OpenidClientScope, error) {
+	var clientScopes []OpenidClientScope
+	var openidClientScopes []*OpenidClientScope
+
+	err := keycloakClient.get(fmt.Sprintf("/realms/%s/client-scopes", realmId), &clientScopes)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, clientScope := range clientScopes {
+		if clientScope.Protocol == "openid-connect" && filter(&clientScope) {
+			func(cs OpenidClientScope) {
```

Comment:
What's with the IIFE here?

Reply:
Good question!  This one had me scratching my head for a while.

The reason for this is because the loop variable `clientScope` is the same memory reference holding a different value for each iteration.  Since I'm appending a pointer to `clientScope` to a slice when my conditional is true, the end result (without the IIFE) is a slice filled with the same pointer to whatever value happens to be last in `clientScopes`.

The IIFE allows me to pass a copy of the value to the function, meaning that a different memory address is stored in the resulting slice.

Reply:
@mrparkers After looking at https://github.com/mrparkers/terraform-provider-keycloak/commit/8bf7dc82bd1de8053c3a7ddbdbe42604318b4187, I think an explicit `new` would be clearer (and could be done conditionally, instead of copying every client scope and not always using it).

IMO, the IIFE and the new `filter` function signature are sort of hiding that a copy has to happen somewhere.

Reply:
Yeah, I support it is idiomatic in Go to not use syntax to hide stuff like this.

Fixed in https://github.com/mrparkers/terraform-provider-keycloak/pull/59/commits/f0941814c92673e41c8c4a7b751c4e8ff38c3c61